### PR TITLE
Implement css vars

### PR DESCRIPTION
**Pull request by tmolitor-stud-tu**
I'm not sure if this complys wih your coding guidlines and it might not be the way you'd like it be solved, but maybe it's still useful :)

### CODE REVIEW

**Review by bsweeney — CHANGES_REQUESTED** [review]
I look forward to reviewing this change more fully. In the meantime a few small requests:

- Across the board you should to wrap the body of a conditional statement, loop, etc. in braces (`{ }`) to avoid potential mistake or misunderstanding during future edits.
- Do you have test cases?

**Comment by tmolitor-stud-tu** [comment]
I've fixed the curly braces, testcases are coming soon :)

**Comment by tmolitor-stud-tu** [comment]
A test case would be the following html code (I'm not sure how to write proper dompdf testcases to test this programmatically):

```
<!DOCTYPE html>
<html>
	<head>
		<style>
			.inner {
				background-color: var(--custom-color);
			}
			.middle1 {
				background-color: var(--custom-color);
			}
			div {
				height: 4em;
				width: 4em;
			}
			.outer {
				height: 16em;
				width: 16em;
				--custom-color: green;
				background-color: blue;
			}
			.middle2 {
				background-color: yellow;
				height: 9em;
				width: 9em;
				--custom-color: red;
			}
		</style>
	</head>
	<body>
		<div class="outer">
			<div class="middle1"></div>
			<div class="middle2">
				<div class="inner"></div>
			</div>
		</div>
	</body>
</html>
```
See [this jsFiddle](https://jsfiddle.net/k40cqfpy/)
And [the PDF output](https://github.com/dompdf/dompdf/files/12552603/dompdf_cssvar.pdf)

**Comment by wundii** [comment]
i think, that including the css vars is a good idea, unfortunately this does not work with bootstrap 5.*, with defined variables in the :root{--bs-green:#198754} element. warning: Undefined array key "__bs_green"

**Comment by bsweeney** [comment]
@tmolitor-stud-tu thoughts on #3314? I think that PR takes the idea a bit further to accommodate, for example, inheritance.

**Comment by bsweeney** [comment]
@tmolitor-stud-tu any thoughts on #3314?

File: src/Css/Style.php
Comment (bsweeney) [inline src/Css/Style.php:1578]: This should probably be handled in the compute_prop method so that the value can then be parsed by any property-specific compute methods.

File: src/Css/Style.php
Comment (bsweeney) [inline src/Css/Style.php:1516]: I think think we want to return here. How would we accommodate something like:

```
<style>
    :root {
        --half: 50%;
    }
    div {
        width: var(--half);
    }
</style>
<div>dompdf</div>
```

File: src/Css/Style.php
Comment (bsweeney) [reply src/Css/Style.php:1578]: Is this logic sufficient to accommodate something like:

```
<style>
    :root {
        --half: 50%;
    }
    div {
        width: var(--half);
    }
</style>
<div>dompdf</div>
```

**Review by bsweeney — CHANGES_REQUESTED** [review]
Some initial thoughts for discussion. I haven't thoroughly evaluated this in relation to the spec nor in relation to how we generally parse CSS properties.

**Comment by bsweeney** [comment]
FYI at this point I'm inclined to go with 3314 since some of the logic is built out a bit more. Maybe you can collaborate on that PR?

**Comment by tmolitor-stud-tu** [comment]
> FYI at this point I'm inclined to go with 3314 since some of the logic is built out a bit more. Maybe you can collaborate on that PR?

That's okay with me :) I'm just glad support for css-vars is finally getting into dompdf. And if my pr did just the initial kickoff for this, that's fine by be, too :)

**Comment by bsweeney** [comment]
I think the other PR is just about ready. Just waiting on any final reviews from involved devs.

**Pull request closed without merging**
